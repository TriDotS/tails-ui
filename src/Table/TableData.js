import styled from 'styled-components'

const TableData = styled.td`
  padding: 10px;
  text-align: center;

  @media(max-width: 600px) {
    display: block;
    text-align: right;
    font-size: 13px;
    border-bottom: 1px dotted #ccc;

    &:last-child {
      border-bottom: 0;
    }

    &:before {
      content: attr(data-label);
      float: left;
      text-transform: uppercase;
      font-weight: bold;
    }
  }
`

export default TableData
