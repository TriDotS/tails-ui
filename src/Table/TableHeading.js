import styled from 'styled-components'

const TableHeading = styled.th`
  padding: 10px;
  text-align: center;
  text-transform: uppercase;
  font-size: 14px;
  letter-spacing: 1px;
`

export default TableHeading
