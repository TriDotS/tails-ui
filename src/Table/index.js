import React from 'react'
import styled from 'styled-components'

const Table = styled.table`
  border: 1px solid #ccc;
  width: 100%;
  margin: 0;
  padding: 0;
  border-collapse: collapse;
  border-spacing: 0;

   @media(max-width: 600px) {
    border: none;
  }
`

const TableHead = styled.thead`
  @media(max-width: 600px) {
    display: none;
  }
`

const TableBody = styled.tbody`
  padding: 0;
`

const TableRow = styled.tr`
  border: 1px solid #ddd;
  padding: 5px;

   @media(max-width: 600px) {
    margin-bottom: 10px;
    display: block;
    border-bottom: 2px solid #ddd;
  }
`

const TableHeading = styled.th`
  padding: 10px;
  text-align: center;
  text-transform: uppercase;
  font-size: 14px;
  letter-spacing: 1px;
`

const TableData = styled.td`
  padding: 10px;
  text-align: center;

   @media(max-width: 600px) {
    display: block;
    text-align: right;
    font-size: 13px;
    border-bottom: 1px dotted #ccc;

    &:last-child {
      border-bottom: 0;
    }

    &:before {
      content: attr(data-label);
      float: left;
      text-transform: uppercase;
      font-weight: bold;
    }
  }
`

export default () =>
  <Table>
    <TableHead>
      <TableRow>
        <TableHeading>Payment</TableHeading>
        <TableHeading>Amount</TableHeading>
        <TableHeading>Total</TableHeading>
      </TableRow>
    </TableHead>
    <TableBody>
      <TableRow>
        <TableData data-label="payment">Payment</TableData>
        <TableData data-label="amount">Amount</TableData>
        <TableData data-label="total">Total</TableData>
      </TableRow>
      <TableRow>
        <TableData data-label="payment">Payment</TableData>
        <TableData data-label="amount">Amount</TableData>
        <TableData data-label="total">Total</TableData>
      </TableRow>
      <TableRow>
        <TableData data-label="payment">Payment</TableData>
        <TableData data-label="amount">Amount</TableData>
        <TableData data-label="total">Total</TableData>
      </TableRow>
      <TableRow>
        <TableData data-label="payment">Payment</TableData>
        <TableData data-label="amount">Amount</TableData>
        <TableData data-label="total">Total</TableData>
      </TableRow>
    </TableBody>
  </Table>
