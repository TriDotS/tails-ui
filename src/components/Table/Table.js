import styled from 'styled-components'

const Table = styled.table`
  border: 1px solid #ccc;
  width: 100%;
  margin: 0;
  padding: 0;
  border-collapse: collapse;
  border-spacing: 0;

  @media (max-width: 600px) {
    border: none;
  }
`
export default Table
