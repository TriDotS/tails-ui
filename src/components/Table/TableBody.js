import styled from 'styled-components'

const TableBody = styled.tbody`padding: 0;`

export default TableBody
