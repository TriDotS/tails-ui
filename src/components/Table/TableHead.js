import styled from 'styled-components'

const TableHead = styled.thead`@media (max-width: 600px) {display: none;}`

export default TableHead
