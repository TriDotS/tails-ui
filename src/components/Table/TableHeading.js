import React from 'react'

const TableHeading = ({ children }) => (
  <th className="p-2 text-center uppercase text-lg tracking-wide">
    {children}
  </th>
)

export default TableHeading
