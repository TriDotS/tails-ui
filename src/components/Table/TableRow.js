import styled from 'styled-components'

const TableRow = styled.tr`
  border: 1px solid #ddd;
  padding: 5px;
  @media (max-width: 600px) {
    margin-bottom: 10px;
    display: block;
    border-bottom: 2px solid #ddd;
  }
`

export default TableRow
