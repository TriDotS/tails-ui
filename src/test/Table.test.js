import React from 'react'
import Table from '../components/Table'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(
    <Table
      cols={[
        { key: 'firstName', label: 'First Name' },
        { key: 'lastName', label: 'Last Name' },
        { key: 'age', label: 'Age' },
        { key: 'weight', label: 'weight' },
        { key: 'dog', label: 'Dog' }
      ]}
      data={[
        {
          id: 1,
          firstName: 'John',
          lastName: 'Doe',
          age: 123,
          weight: 1234,
          dog: 234234
        },
        {
          id: 2,
          firstName: 'John',
          lastName: 'Doe',
          age: 123,
          weight: 1234,
          dog: 234234
        },
        {
          id: 3,
          firstName: 'John',
          lastName: 'Doe',
          age: 123,
          weight: 1234,
          dog: 234234
        },
        {
          id: 4,
          firstName: 'John',
          lastName: 'Doe',
          age: 123,
          weight: 1234,
          dog: 234234
        },
        {
          id: 5,
          firstName: 'Clark',
          lastName: 'Kent',
          age: 12938,
          weight: 1345235,
          dog: 1231230980980089080890809080898123
        }
      ]}
    />,
    div
  )
})

test('table snapshot', () => {
  const tree = renderer
    .create(
      <Table
        cols={[
          { key: 'firstName', label: 'First Name' },
          { key: 'lastName', label: 'Last Name' },
          { key: 'age', label: 'Age' },
          { key: 'weight', label: 'weight' },
          { key: 'dog', label: 'Dog' }
        ]}
        data={[
          {
            id: 1,
            firstName: 'John',
            lastName: 'Doe',
            age: 123,
            weight: 1234,
            dog: 234234
          },
          {
            id: 2,
            firstName: 'John',
            lastName: 'Doe',
            age: 123,
            weight: 1234,
            dog: 234234
          },
          {
            id: 3,
            firstName: 'John',
            lastName: 'Doe',
            age: 123,
            weight: 1234,
            dog: 234234
          },
          {
            id: 4,
            firstName: 'John',
            lastName: 'Doe',
            age: 123,
            weight: 1234,
            dog: 234234
          },
          {
            id: 5,
            firstName: 'Clark',
            lastName: 'Kent',
            age: 12938,
            weight: 1345235,
            dog: 1231230980980089080890809080898123
          }
        ]}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
