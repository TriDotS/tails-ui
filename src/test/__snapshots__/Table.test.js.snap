// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table snapshot 1`] = `
<div
  className="overflow-x-scroll"
>
  <table
    className="w-full border-2 border-grey-light"
  >
    <thead>
      <tr
        className="border-b-2"
      >
        <th
          className="text-left p-2"
        >
          First Name
        </th>
        <th
          className="text-left p-2"
        >
          Last Name
        </th>
        <th
          className="text-left p-2"
        >
          Age
        </th>
        <th
          className="text-left p-2"
        >
          weight
        </th>
        <th
          className="text-left p-2"
        >
          Dog
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="border-b-2"
      >
        <td
          className="text-left p-2"
        >
          John
        </td>
        <td
          className="text-left p-2"
        >
          Doe
        </td>
        <td
          className="text-left p-2"
        >
          123
        </td>
        <td
          className="text-left p-2"
        >
          1234
        </td>
        <td
          className="text-left p-2"
        >
          234234
        </td>
      </tr>
      <tr
        className="border-b-2"
      >
        <td
          className="text-left p-2"
        >
          John
        </td>
        <td
          className="text-left p-2"
        >
          Doe
        </td>
        <td
          className="text-left p-2"
        >
          123
        </td>
        <td
          className="text-left p-2"
        >
          1234
        </td>
        <td
          className="text-left p-2"
        >
          234234
        </td>
      </tr>
      <tr
        className="border-b-2"
      >
        <td
          className="text-left p-2"
        >
          John
        </td>
        <td
          className="text-left p-2"
        >
          Doe
        </td>
        <td
          className="text-left p-2"
        >
          123
        </td>
        <td
          className="text-left p-2"
        >
          1234
        </td>
        <td
          className="text-left p-2"
        >
          234234
        </td>
      </tr>
      <tr
        className="border-b-2"
      >
        <td
          className="text-left p-2"
        >
          John
        </td>
        <td
          className="text-left p-2"
        >
          Doe
        </td>
        <td
          className="text-left p-2"
        >
          123
        </td>
        <td
          className="text-left p-2"
        >
          1234
        </td>
        <td
          className="text-left p-2"
        >
          234234
        </td>
      </tr>
      <tr
        className="border-b-2"
      >
        <td
          className="text-left p-2"
        >
          Clark
        </td>
        <td
          className="text-left p-2"
        >
          Kent
        </td>
        <td
          className="text-left p-2"
        >
          12938
        </td>
        <td
          className="text-left p-2"
        >
          1345235
        </td>
        <td
          className="text-left p-2"
        >
          1.2312309809800891e+33
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
